src/main/java/bio/terra/service/resourcemanagement/azure/AzureContainerPdao.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.service.resourcemanagement.azure;
+
+import bio.terra.common.exception.PdaoException;
+import bio.terra.model.BillingProfileModel;
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import com.azure.storage.file.datalake.DataLakeServiceClient;
+import com.azure.storage.file.datalake.models.DataLakeStorageException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AzureContainerPdao {
+
+  private final AzureAuthService authService;
+
+  @Autowired
+  public AzureContainerPdao(AzureAuthService authService) {
+    this.authService = authService;
+  }
+
+  /**
+   * Get or create a container (also known as FileSystem) in an Azure storage account
+   *
+   * @param profileModel The profile that describes information needed to access the storage account
+   * @param storageAccountResource Metadata describing the storage account that contains the
+   *     container
+   * @param containerType The nature of the container
+   * @return A client connection object that can be used to create folders and files
+   */
+  public DataLakeFileSystemClient getOrCreateContainer(
+      BillingProfileModel profileModel,
+      AzureStorageAccountResource storageAccountResource,
+      AzureStorageAccountResource.ContainerType containerType) {
+    DataLakeServiceClient dataLakeClient =
+        authService.getDataLakeClient(profileModel, storageAccountResource);
+
+    try {
+      DataLakeFileSystemClient fileSystemClient =
+          dataLakeClient.getFileSystemClient(
+              storageAccountResource.determineContainer(containerType));
+      // This will 404 if the container doesn't exist
+      fileSystemClient.getProperties();
```

Comment:
Seems like a hack; I guess there's no better way to do this?

Reply:
Yeah, I agree...there's no `exists` type function...short of doing a list request but even that...there's no find by exact match so you end up having to do some streaming logic.  This seemed less gross